src/Features/CSharp/Portable/ExtractMethod/CSharpSelectionResult.cs

```diff
@@ -71,15 +72,16 @@ internal abstract partial class CSharpSelectionResult : SelectionResult
         {
         }
 
-        protected override bool UnderAsyncAnonymousMethod(SyntaxToken token, SyntaxToken firstToken, SyntaxToken lastToken)
+        protected override bool UnderAnonymousOrLocalMethod(SyntaxToken token, SyntaxToken firstToken, SyntaxToken lastToken)
         {
             var current = token.Parent;
             for (; current != null; current = current.Parent)
             {
                 if (current is MemberDeclarationSyntax ||
                     current is SimpleLambdaExpressionSyntax ||
                     current is ParenthesizedLambdaExpressionSyntax ||
-                    current is AnonymousMethodExpressionSyntax)
+                    current is AnonymousMethodExpressionSyntax ||
+                    current is LocalFunctionStatementSyntax)
```

Comment:
tests added for this?

Reply:
Indirectly Yes. It's the first test added in the cshsrp tests.

Reply:
#resolved thanks